include/simdjson/dom/parser.h

```diff
@@ -114,8 +114,30 @@ class parser {
    * The buffer must have at least SIMDJSON_PADDING extra allocated bytes. It does not matter what
    * those bytes are initialized to, as long as they are allocated.
    *
-   * If realloc_if_needed is true, it is assumed that the buffer does *not* have enough padding,
-   * and it is copied into an enlarged temporary buffer before parsing.
+   * If realloc_if_needed is true (the default), it is assumed that the buffer does *not* have enough padding,
+   * and it is copied into an enlarged temporary buffer before parsing. Thus the following is safe: 
+   * 
+   *   const char *json      = "{\"key\":\"value\"}";
+   *   const size_t json_len = strlen(json);
+   *   simdjson::dom::parser parser;
+   *   simdjson::dom::element element = parser.parse(json, json_len);
+   * 
+   * If you set realloc_if_needed to false (e.g., parser.parse(json, json_len, false)), 
+   * you must provide a buffer with at least SIMDJSON_PADDING extra bytes at the end.
+   * The benefit of setting realloc_if_needed to false is that you avoid a temporary
+   * memory allocation and a copy.
+   * 
+   * The padded bytes may be read. It is not important how you initialize
+   * these bytes though we recommend a sensible default like null character values or spaces.
+   * For example, the following low-level code is safe:
+   * 
+   *   const char *json      = "{\"key\":\"value\"}";
+   *   const size_t json_len = strlen(json);
+   *   std::unique_ptr<char[]> padded_json_copy{new char[json_len + SIMDJSON_PADDING]};
+   *   memcpy(padded_json_copy.get(), json, json_len);
+   *   memset(padded_json_copy.get() + json_len, 0, SIMDJSON_PADDING);
```

Comment:
Could perhaps '\0' or ' ' be used instead of 0, to really connect the code to the suggestion in the text?

Reply:
Done.